payments-core/src/main/java/com/stripe/android/payments/paymentlauncher/PaymentLauncherViewModel.kt

```diff
@@ -77,48 +85,70 @@ internal class PaymentLauncherViewModel @Inject constructor(
      * this value will be set true, preventing [PaymentLauncherConfirmationActivity] to try to
      * confirm the same [StripeIntent] again.
      */
-    internal val hasStarted: Boolean
+    private val hasStarted: Boolean
         get() = savedStateHandle.get(KEY_HAS_STARTED) ?: false
 
+    /**
+     * Set when activity calls [register]. Used to redirect back to calling activity.
+     */
+    private var authActivityStarterHost: AuthActivityStarterHost? = null
```

Comment:
This host an activity internally. A viewModel cannot hold a reference to an Activity / any other view.

The Activity / Fragment gets destroyed and recreated on every configuration change , and a new instance gets the same instance of the ViewModel. Keeping a reference to the activity in the ViewModel leads to memory leaks.

Reply:
Update on this - Looks like we were already leaking by keeping an activity reference on the viewModel, so this seems out of the scope of this bugfix. Unblocking.